flax/core/scope.py

```diff
@@ -191,11 +188,43 @@ def group_collections(xs: VariableDict,
   return tuple(groups)
 
 
+T = TypeVar('T')
```

Comment:
Why move this back to scope? It was intentionally placed in a separate file so that we can refer to its documentation separately from the rest of scope. Users using the Module API directly will need to know about variables (thus may want to look at variable.py) but don't need to know about scopes.

Reply:
The code was incorrect it had a reference 'Scope' but it wasn't imported. But importing creates a circular dependency.